File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/rust_solutions/349-intersection-of-two-arrays.rs
use std::collections::HashSet;

impl Solution {
    pub fn intersection(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut set1: HashSet<i32> = nums1.into_iter().collect();
        let mut result_set: HashSet<i32> = HashSet::new();

        for num in nums2 {
            if set1.contains(&num) {
                result_set.insert(num);
            }
        }

        result_set.into_iter().collect()
    }
}

File: leetcode/AIcoding/3.DeepSeek-v3/go_solutions/349-intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {
    set := make(map[int]bool)
    result := make([]int, 0)

    for _, num := range nums1 {
        set[num] = true
    }

    for _, num := range nums2 {
        if set[num] {
            result = append(result, num)
            delete(set, num)
        }
    }

    return result
}